plugins/chromatic_correction.py

```diff
@@ -0,0 +1,252 @@
+                        "Pixel size cor": [float(scale[0]), float(scale[1])],
+                        "Centre position cor": [float(translation[0]), float(translation[1])],
+                        "Rotation cor": float(rotation),
+                        "Shear cor": float(shear),
+                        # Extra information saved for tracking/debugging purpose
+                        "tint": list(im.metadata[model.MD_USER_TINT]),
+                        "excitation": [float(input_wl[0]), float(input_wl[1])],
+                        "emission": [float(output_wl[0]), float(output_wl[1])],
+                        "exposure time": float(im.metadata[model.MD_EXP_TIME]),
+                        "light power": float(im.metadata[model.MD_LIGHT_POWER])
+                    }
+
+                    # update metadata
+                    im.metadata.update({model.MD_PIXEL_SIZE_COR: scale})
+                    im.metadata.update({model.MD_POS_COR: translation})
+                    im.metadata.update({model.MD_ROTATION_COR: rotation})
+                    im.metadata.update({model.MD_SHEAR_COR: shear})
+
+                    # save the image and display
+                    self.show_static_stream(f"modified_{im.metadata[model.MD_DESCRIPTION]}_{i}.ome.tiff", im)
+                except (ValueError, KeyError) as ex:
+                    logging.error("Failed to compute chromatic aberration correction yaml file due to: %s", ex)
+
+        return trans_per_channel
+
+    def start(self):
+        localization_tab = self.main_app.main_data.getTabByName("cryosecom-localization")
+        tab_data = localization_tab.tab_data_model
+        other_mip_streams = []
+        reference_mip_stream = []
+
+        if len(tab_data.streams.value) <= 1:
+            box = wx.MessageDialog(self.main_app.main_frame,
+                                   "Add minimum two streams. One of the streams must be green channel.",
+                                   "Failed to do chromatic correction", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        levels = generate_zlevels(tab_data.main.focus,
+                                  (-10e-06, 10e-06),
+                                  5e-06)
+        non_static = [s for s in tab_data.streams.value
+                      if isinstance(s, FluoStream)]
+        zlevels = {s: levels for s in non_static}
+        settings_observer = tab_data.main.settings_obs
+        acqui_task = acquireZStack(non_static, zlevels, settings_obs=settings_observer)
+        das, e = acqui_task.result()
+
+        # Calculate Maximum Intensity projection for all the present channels in the stream panel
+        for i, da in enumerate(das):
+            mip_image = model.DataArray(numpy.amax(da, axis=0), copy.copy(da.metadata))
+            dataio.tiff.export(f"{da.metadata[model.MD_DESCRIPTION]}_{i}.ome.tiff", mip_image)
```

Comment:
As each stream has a different name, I think it'd be easier to find the files if you *start* the filename with `{i}`.

Reply:
Would it be a good idea to put datetime stamp instead of simple {i}? It would be indeed nice to start with i for proper sorting of multiple files. Then even if multiple streams have same name, that should not be an issue and it would be fine not to check it, right?